app/controllers/ops_controller/ops_rbac.rb

```diff
@@ -349,9 +349,17 @@ def rbac_tenants_list
   def rbac_tenant_delete
     assert_privileges("rbac_tenant_delete")
     tenants = []
+    projects = []
     if !params[:id] # showing a tenants list
-      ids = find_checked_items.collect { |r| from_cid(r.split("-").last) }
-      tenants = Tenant.find_all_by_id(ids)
+      ids = find_checked_items
+      recs_to_delete = Tenant.find_all_by_id(ids)
+      recs_to_delete.each do |rec|
+        if rec.divisible
+          tenants.push(rec)
+        else
+          projects.push(rec)
+        end
+      end
```

Comment:
sorry - I know this isn't your code but this is looking up all the parent records just to detect if it is a root node.

```ruby
tenants.reject! do |t|
  if t.root?
    add_flash(_("Default %{model} \"%{name}\" can not be deleted") % {:model => ui_lookup(:model => "Tenant"), :name => t.name}, :error)
    # true
  end
end
```

I think calling `add_flash` returns a value, so not having that `true` in there will work. it just seems fragile, so you may want to explicitly set it - up to you